### DISCUSSION TITLE
What is the syntax for FoundryUp version revert?

### DISCUSSION THREAD

**Discussion by 0xFloop** [foundry-rs/foundry#2557] [Help]
Hello, I know this is a fairly elementary issue but I cant seem to find the version syntax for reverting your foundry version to prior releases using the `foundryup -v` command. Thank you to anyone who can provide help. :)

**Comment by onbjerg** [comment]
Hi! It's not super intuitive yet since we don't have actual version numbers yet.

To revert to a previous nightly, the syntax is `foundryup -v nightly-$SHA`, so for example if you wanted to revert to the nightly at commit `9ed1c37f49de428a3c653a343c3ffab0dd7c3790`, you would execute `foundryup -v nightly-9ed1c37f49de428a3c653a343c3ffab0dd7c3790`.

You can find the commit SHA on the releases tab of the repository, but please keep in mind that we only keep 3 nightlies currently.

**Reply by 0xFloop** [reply]
You're a legend, thank you for the help!

**Reply by 0xFloop** [reply]
Running the command you mentioned does install the nightly foundry version specified, but then proceeds to install the latest forge, cast and anvil opposed to the versions from that nightly build. Maybe my initial question was wrong. Im looking to revert to previous forge releases to compare testing times/outputs (which i maybe wrongly assumed would be done by reverting foundry version).  I've attached an image of the terminal output. Thanks again for any help :)


![Screen Shot 2022-08-03 at 10 47 57 AM](https://user-images.githubusercontent.com/95703085/182675266-29dd53e8-32c8-4cc1-960e-11b5f9b93490.png)

**Reply by onbjerg** [reply]
That nightly version does not exist. The reason it looks like it is installing the latest version is that it just blindly does `forge --version` after. It *should* catch the error though, so try updating `foundryup`.

But yeah, as mentioned, that nightly does not exist, we only keep 3 nightlies. If you have Rust installed you can use `-C <sha>` to install any commit, but it will take a while since it clones the repository and builds the project for you :)

**Reply by 0xFloop** [reply]
![Screen Shot 2022-08-03 at 11 27 52 AM](https://user-images.githubusercontent.com/95703085/182682244-81ae2d07-3b90-41ee-9ab1-8d4a823cb814.png) I see now, i just assumed i could use it since its on the releases page. Thank you for the help.

**Reply by onbjerg** [reply]
That's the tags page, the release does not exist anymore, so there are no prebuilt binaries: https://github.com/foundry-rs/foundry/releases

**Reply by 0xFloop** [reply]
oops im an idiot, thank you for the help and sorry for the confusion
